### PR TITLE
manually attempt connection / simulate a rtt

### PR DESCRIPTION
**Pull request by willscott**
First pass at a latency probe

* we should see if this attempt at reading (the only thing that actually blocks down low enough to need a libp2p rtt) and then closing is a reasonable way to pull out an RTT latency
* we should make sure this can't error in unexpected ways.

### CODE REVIEW

File: netutil/netutil.go [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+			end = time.Now()
+		} else {
+			err = serr
+		}
+	}()
+
+	// send a write of random bytes, which should make the other side unhappy.
+	stream.Write([]byte("-latencytest-\n\x00"))
+	wg.Wait()
+	defer conn.Close()
```
Comment (gammazero) [inline netutil/netutil.go:50]: Does not seem to be a need to defer here.

File: netutil/netutil.go [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
```
Comment (gammazero) [inline netutil/netutil.go:51]:
```go
if err != nil {
   return nil, 0, err
}
```

**Review by gammazero — CHANGES_REQUESTED** [review]

**Review by gammazero — APPROVED** [review]

File: netutil/netutil.go (hunk 2 above)
Comment (willscott) [reply netutil/netutil.go:51]: we may still get a useful remote node multiaddr by this point, which is why i'm tempted to still return it even if the error indicates the measured latency shouldn't be trusted

**Review by hannahhoward — APPROVED** [review]
LGTM

**Pull request merged**